### PR TITLE
Ol3 coding style

### PR DESCRIPTION
**Pull request by ger-benjamin**
Fix: https://github.com/camptocamp/ngeo/issues/446

Examples:
 - http://ger-benjamin.github.io/ngeo/ol3_coding_style/examples (choose some)
 - http://ger-benjamin.github.io/ngeo/ol3_coding_style/examples/contribs/gmf (choose some)

### CODE REVIEW

File: contribs/gmf/src/directives/authenticationdirective.js [hunk 1]
```diff
@@ -225,17 +225,11 @@ gmf.AuthenticationController.prototype.resetPassword = function() {
...
-           * @param {gmf.AuthenticationDefaultResponse} respData Response.
-           */
-          function(respData) {
-            this.resetPasswordModalShown = true;
-            this.resetError_();
-          },
-          this
-      ),
-      goog.bind(this.setError_, this, error)
+    function(respData) {
```
Comment (gberaudo) [inline contribs/gmf/src/directives/authenticationdirective.js]: The function type should be kept.

File: contribs/gmf/src/directives/themeselector.js [hunk 2]
```diff
@@ -142,10 +142,7 @@ gmf.ThemeselectorController.prototype.setLocationPath_ = function(themeId) {
  * @private
  */
 gmf.ThemeselectorController.prototype.setThemes_ = function() {
-  this.gmfThemes_.getThemesObject().then(goog.bind(
-      /**
-       * @param {Array.<Object>} themes Array of theme objects.
-       */
+  this.gmfThemes_.getThemesObject().then(
```
Comment (gberaudo) [inline contribs/gmf/src/directives/themeselector.js]: Idem https://github.com/camptocamp/ngeo/pull/774/files#r54395955.

**Comment by ger-benjamin** [comment]
Thanks @gberaudo . I don't see your post sooner, so there are 2 or 3 new removal of 'function type'. I'll correct that soon.

**Comment by ger-benjamin** [comment]
In the last commit, I've removed all the "goog/angular.bind" functions except 6 with an annotation over the function (f.i: https://github.com/camptocamp/ngeo/blob/master/contribs/gmf/src/directives/themeselector.js#L145-L167)

**Comment by gberaudo** [comment]
@ger-benjamin, it is technically possible to have both `bind` and annotations. See https://github.com/gberaudo/ngeo/commit/2cd08ac89467a47c38b6da9e2c9d3bea99025b4a.

**Comment by ger-benjamin** [comment]
Done, thanks @gberaudo 
Example up to date.

File: contribs/gmf/src/controllers/abstract.js [hunk 3]
```diff
@@ -123,7 +123,7 @@ gmf.AbstractController = function(config, $scope, $injector) {
 
   /**
    * Languages URL
-   * @type {Object.<string, string>}
+   * @type {Object!}
```
Comment (gberaudo) [inline contribs/gmf/src/controllers/abstract.js]: The `!` should be used as a prefix.
What is the reason for removing the `<string, string>` typing?

Comment (ger-benjamin) [inline contribs/gmf/src/controllers/abstract.js]: Object.key want an @type {Object} (not null)...

Comment (gberaudo) [inline contribs/gmf/src/controllers/abstract.js]: Sure. What about using `@type {!Object.<string, string>}`?

File: contribs/gmf/src/controllers/abstract.js [hunk 4]
```diff
@@ -123,7 +123,7 @@ gmf.AbstractController = function(config, $scope, $injector) {
 
   /**
    * Languages URL
-   * @type {Object.<string, string>}
```
Comment (fredj) [inline contribs/gmf/src/controllers/abstract.js]: `@type {!Object.<string, string>}`

**Comment by fredj** [comment]
+1

File: contribs/gmf/src/controllers/abstract.js (hunk 4 above)
Comment (ger-benjamin) [inline contribs/gmf/src/controllers/abstract.js]: Done, thanks

**Comment by ger-benjamin** [comment]
Last self-check: ok.
I merge.

**Pull request merged**